odcockpit/src/plugins/environmentviewer/EnvironmentViewerGLWidget.cpp

```diff
@@ -442,6 +449,49 @@ namespace cockpit {
                         glEnd();
                     }
     */
+                    //Visualize compact point cloud, where points are sorted by increasing azimuth (increment the azimuth after every 16 points) and 
+                    //vertical angle (from -15 to 15 with increment 2 for each 16 points).
+                    //A compact point cloud contains: (1) the starting azimuth, (2) the ending azimuth, (3) number of points per azimuth, 
+                    //and (4) a string with the distance values of all points in the container
+                    if (m_CPCreceived) {
+                        Lock lockCPC(m_cpcMutex);
+                        float startAzimuth = m_cpc.getStartAzimuth();
+                        float endAzimuth = m_cpc.getEndAzimuth();
+                        uint8_t entriesPerAzimuth = m_cpc.getEntriesPerAzimuth();
+                        string distances = m_cpc.getDistances();
+                        uint32_t numberOfPoints = distances.size()/2;
+                        uint32_t numberOfAzimuths = numberOfPoints/entriesPerAzimuth;
+                        float azimuthIncrement = (endAzimuth-startAzimuth)/numberOfAzimuths;//Calculate the azimuth increment
+                        stringstream sstr(distances);
+                        
+                        m_root->render(m_renderingConfiguration); 
+                        glPushMatrix();
+                        glPointSize(1.0f); //set point size to 1 pixel
+                        glBegin(GL_POINTS); //starts drawing of points
+                        glColor3f(255.0f,255.0f,0.0);//Yellow color
+                        
+                        static half distance_h(0.0);
+                        static float xyDistance, xData, yData, zData;
+                        float azimuth = startAzimuth;             
+                        for (uint32_t azimuthIndex = 0; azimuthIndex < numberOfAzimuths; azimuthIndex++) {
+                            float verticalAngle = START_V_ANGLE;
+                            for (uint8_t sensorIndex = 0; sensorIndex<entriesPerAzimuth; sensorIndex++) {
+                                sstr.read((char*)(&distance_h), 2);//Read distance value from the string in a CPC container point by point
+                                float distance = static_cast<float>(distance_h);
+                                //Compute x, y, z coordinate based on distance, azimuth, and vertical angle
+                                xyDistance = distance*cos(toRadian(verticalAngle));
```

Comment:
toRadian() can be turned by multiplying a plain const:

`const float DEG_TO_RAD = M_PI/180.0;`